cmake/Dependencies.cmake

```diff
@@ -916,17 +916,20 @@ endif()
 
 # ---[ NCCL
 if(USE_NCCL)
-  if(NOT USE_CUDA)
+  if(NOT (USE_CUDA OR USE_ROCM))
```

Comment:
`if(NOT USE_GPU)`?

Additionally, this branch may not even be required, because you have already made `USE_NCCL` a build option that depends on `USE_GPU`.

Reply:
Would you like me to remove the branch or leave that for a later date?

Reply:
Either way is fine. To push forward this PR faster, it's OK if you leave the branch here.

Reply:
@xuhdev I don't think the `USE_GPU` variable from `CMakeLists.txt` is visible to the `cmake/Dependencies.cmake` file, because I'm seeing `USE_NCCL=ON` in non-CUDA/ROCm CI builds, causing failures. One example: https://circleci.com/gh/pytorch/pytorch/2450228

Reply:
@jithunnair-amd Is it because `USE_ROCM` is `ON` by default? Note that `USE_CUDA` has a lot of preprocessing in Python scripts but `USE_ROCM` is now purely done in CMake.

Reply:
@xuhdev Is it okay to set default value of `USE_ROCM` in `CMakeLists.txt` to `OFF` in that case? I think the `.jenkins/[pytorch|caffe2]/build.sh` scripts set it to 1 for any ROCm builds.

Reply:
It should be fine. Any external settings will override the default.

Reply:
For my edification, how exactly does USE_ROCM get set to OFF for any non-ROCm builds?